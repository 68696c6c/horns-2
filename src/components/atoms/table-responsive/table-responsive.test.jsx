/* global describe, it, expect */
import React from 'react'

import { render } from 'setup-test'

import TableResponsive from '.'

describe('TableResponsive', () => {
  it('should render as default', () => {
    const { container } = render(
      <TableResponsive
        minWidth="1200px"
        rowData={[
          { id: '1', name: 'first', email: 'first@example.com' },
          { id: '2', name: 'second', email: 'second@example.com' },
        ]}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
