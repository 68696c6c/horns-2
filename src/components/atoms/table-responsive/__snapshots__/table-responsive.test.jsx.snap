// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableResponsive should render as default 1`] = `
.emotion-1 {
  width: 100%;
  height: auto;
  overflow-x: scroll;
}

.emotion-0 {
  width: 100%;
  border-collapse: collapse;
}

.emotion-0 td,
.emotion-0 th {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 8px;
  padding-right: 8px;
}

.emotion-0 th {
  font-family: Helvetica,sans-serif;
  font-size: medium;
  font-style: normal;
  font-weight: 700;
  line-height: 1em;
}

.emotion-0 td {
  font-family: Helvetica,sans-serif;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1em;
}

.emotion-0 thead {
  display: none;
}

.emotion-0 tbody:nth-child(odd) {
  background: rgb(235,235,235);
  color: rgb(1,1,1);
  border-color: rgb(216,216,216);
}

@media (min-width:480px) {
  .emotion-0 {
    min-width: 1200px;
  }

  .emotion-0 thead {
    display: table-header-group;
  }

  .emotion-0 tbody {
    display: table-row;
  }

  .emotion-0 tbody tr {
    display: table-cell;
  }

  .emotion-0 tbody tr th {
    display: none;
  }

  .emotion-0 tbody tr td {
    display: block;
  }
}

<div>
  <div
    class="emotion-1"
    height="auto"
    x="scroll"
  >
    <table
      class="emotion-0"
    >
      <thead>
        <tr>
          <th>
            id
          </th>
          <th>
            name
          </th>
          <th>
            email
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th>
            id
          </th>
          <td>
            1
          </td>
        </tr>
        <tr>
          <th>
            name
          </th>
          <td>
            first
          </td>
        </tr>
        <tr>
          <th>
            email
          </th>
          <td>
            first@example.com
          </td>
        </tr>
      </tbody>
      <tbody>
        <tr>
          <th>
            id
          </th>
          <td>
            2
          </td>
        </tr>
        <tr>
          <th>
            name
          </th>
          <td>
            second
          </td>
        </tr>
        <tr>
          <th>
            email
          </th>
          <td>
            second@example.com
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
